### PR TITLE
need a fast get current seqNum method

### PR DESCRIPTION
**Pull request by yingjieMiao**
@eishay @ymatsuda @andrewconner 

We have incrementAndGet() method for sequenceNumber. 
For fast reindex, search need to query the current sequence number, i.e. the last generated one.

### CODE REVIEW

File: kifi-backend/modules/sqldb/app/com/keepit/common/db/slick/H2.scala [hunk 1]
```diff
@@ -33,6 +33,13 @@ class H2(val masterDb: SlickDatabase, val slaveDb: Option[SlickDatabase])
       rs.next()
       SequenceNumber(rs.getLong(1))
     }
+
+    def getLastGeneratedSeq()(implicit session: RSession): SequenceNumber = {
+      val stmt = session.conn.prepareStatement(s"""SELECT CURRVAL('$name')""")
```
Comment (eishay) [inline kifi-backend/modules/sqldb/app/com/keepit/common/db/slick/H2.scala:38]: look at @andrewconner 's code from the weekend using 

```
sql"""SELECT CURRVAL('#$name')"""
```

Should do the same trick. 
Is it right @andrewconner ?

**Comment by eishay** [comment]
nice!
the latest generated id may not be persisted in the model table, but i assume it doesn't matter much for that usecase

File: kifi-backend/modules/sqldb/app/com/keepit/common/db/slick/H2.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/sqldb/app/com/keepit/common/db/slick/H2.scala:38]: Yep, should be exactly the same. Both create a prepared statement with no parameters.

**Comment by yingjieMiao** [comment]
Thanks for the input. Merging...

**Comment by eishay** [comment]
will you update the prepareStatement to use the slick syntax?

**Comment by yingjieMiao** [comment]
i will

**Pull request merged**